Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla40161.cs

```diff
@@ -0,0 +1,140 @@
+
+			var imageA = "seth.png";
+			var imageB = "test.jpg";
+
+			var image = new Image()
+			{
+				Source = imageA,
+				Aspect = Aspect.AspectFill,
+
+				// Children of an AbsoluteLayout can potentially inherit its LayoutConstraint.
+				// This should happen if the child H/V options are also set to Fill AND its size
+				// is all proportional. In that case the child fills the size allocated to it by 
+				// the layout and so it's size should be re-computed iff the layout's size has
+				// changed. This behavior is achived by inheriting the layout's LayoutConstraint.
+
+				// *IF* however the H/V options are Center then the Image should be rendered to 
+				// to *AT MOST* the image size regardless of whether the region allocated by
+				// the absolute layout is larger than the image. 
+				HorizontalOptions = LayoutOptions.Center,
+				VerticalOptions = LayoutOptions.Center,
+			};
+
+			AbsoluteLayout.SetLayoutFlags(image, AbsoluteLayoutFlags.All);
+			AbsoluteLayout.SetLayoutBounds(image, new Rectangle(0, 0, 1, 1));
+			absolute.Children.Add(image);
+
+			var stack = new StackLayout();
+			stack.Children.Add(absolute);
+
+			bool flipSwap = false;
+			var swap = new Button() { Text = "SWAP" };
+			swap.Clicked += (object sender, EventArgs e) => 
+			{
+				if (flipSwap)
+					image.Source = imageA;
+				else
+					image.Source = imageB;
+
+				flipSwap = !flipSwap;
+			};
+			stack.Children.Add(swap);
+
+			bool flipLayout = false;
+			var layout = new Button() { Text = "LAYOUT" };
+			layout.Clicked += (object sender, EventArgs e) => 
+			{
+				if (flipLayout)
+				{
+					image.HorizontalOptions = LayoutOptions.Center;
+					image.VerticalOptions = LayoutOptions.Center;
+				}
+				else
+				{
+					image.HorizontalOptions = LayoutOptions.Fill;
+					image.VerticalOptions = LayoutOptions.Fill;
+				}
+
+				flipLayout = !flipLayout;
+			};
+			stack.Children.Add(layout);
+
+			var counter = new Label() { Text = "counter", AutomationId="counter" };
+			var height = new Label() { Text = "height", AutomationId="height" };
+			var width = new Label() { Text = "width", AutomationId= "width" };
+			stack.Children.Add(counter);
+			stack.Children.Add(height);
+			stack.Children.Add(width);
+
+			var count = 0;
+			var refresh = new Button() { Text = "REFRESH" };
+			refresh.Clicked += (object sender, EventArgs e) =>
+			{
+				height.Text = $"h={image.Height}";
+				width.Text = $"w={image.Width}";
+				counter.Text = $"step={count++}";
+			};
+			stack.Children.Add(refresh);
+
+			Content = stack;
+		}
+
+#if UITEST
+		[Test]
+		public void Issue1Test ()
+		{
+			RunningApp.Screenshot ("I am at Issue 40161");
+			RunningApp.WaitForElement (q => q.Marked ("REFRESH"));
+			RunningApp.Screenshot ("I see the first image");
+
+			RunningApp.Tap ("SWAP");
+			RunningApp.Tap ("REFRESH");
+			RunningApp.WaitForElement(q => q.Marked("step=0"));
+			RunningApp.Screenshot ("I swap the image");
+			RunningApp.WaitForElement(q => q.Marked("w=50"));
```

Comment:
The seth image width is not 50. So if it changed to 50 then the size has been recomputed.